idseq_dag/engine/pipeline_step.py

```diff
@@ -96,6 +101,25 @@ def uploading_results(self):
             idseq_dag.util.s3.upload_folder_with_retries(f, self.s3_path(f))
         self.status = StepStatus.UPLOADED
 
+    def update_status_json_file(self):
+        log.write(f"Updating status file for step {self.name}")
+        #  First, update own status dictionary
+        if not "description" in self.status_dict:
+            self.status_dict["description"] = self.step_description()
+        self.status_dict["status"] = self.status
+        if self.input_file_error:
+            self.status_dict["error"] = self.input_file_error.name
+
+        # Then, update file by reading the json, modifying, and overwriting.
+        self.step_status_lock.acquire()
```

Comment:
since you're acquiring and releasing here, it's actually safer to use a `with` context because this doesn't guarantee that release() would be called if there's an exception in between: https://dbader.org/blog/python-context-managers-and-with-statement